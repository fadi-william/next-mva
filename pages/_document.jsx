import Document, { Head, Main, NextScript } from "next/document";

const isProduction = process.env.NODE_ENV === "production";

export default class MyDocument extends Document {
  render() {
    return (
      <html>
        <Head>
          {isProduction && (
            <link rel="stylesheet" href="/_next/static/style.css" />
          )}
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </html>
    );
  }
}
