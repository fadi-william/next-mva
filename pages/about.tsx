// The imported libs.
import * as React from "react";

// The imported external components.
import Layout from "../app/components/layout/Layout/Layout";

export default () => (
  <Layout>
    <p>This is the about page!</p>
  </Layout>
);
