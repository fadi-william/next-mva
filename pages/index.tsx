// The imported libs.
import * as React from "react";

// The imported external components.
import pageWithIntl from "../app/components/hoc/PageWithIntl";
import Layout from "../app/components/layout/Layout/Layout";
import ShowList from "../app/components/lists/show/ShowList/ShowList";

// The mst store.
import { getStore } from "../app/models/show/ShowList";

// The Index's props interface.
interface IIndexProps {
  isServer: boolean;
}

export class Index extends React.Component<IIndexProps> {
  public static getInitialProps({ req }): any {
    const isServer = !!req;
    return {
      isServer
    };
  }

  public render() {
    const { isServer } = this.props;

    const store = getStore(isServer);

    return (
      <Layout>
        <ShowList showList={store} />
      </Layout>
    );
  }
}

export default pageWithIntl(Index);
