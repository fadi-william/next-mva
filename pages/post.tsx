// The imported libs.
import React from "react";

// The imported external components.
import Show from "../app/components/entities/Show/Show";
import pageWithIntl from "../app/components/hoc/PageWithIntl";
import Layout from "../app/components/layout/Layout/Layout";

// The mst store.
import { getStore } from "../app/models/show/Show";

// The show page's props interface.
interface IPostProps extends InjectedIntlProps {
  url: any;
  isServer: boolean;
}

export class Post extends React.Component<IPostProps> {
  public static getInitialProps({ req }) {
    const isServer = !!req;

    return {
      isServer
    };
  }

  public render() {
    const { isServer } = this.props;

    const { id } = this.props.url.query;

    // Set the id of the current show.
    const store = getStore(isServer, Number(id));

    return (
      <Layout>
        <Show show={store} />
      </Layout>
    );
  }
}

export default pageWithIntl(Post);
