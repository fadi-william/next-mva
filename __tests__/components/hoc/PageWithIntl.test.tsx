import * as React from "react";
import * as renderer from "react-test-renderer";

import pageWithIntl from "../../../app/components/hoc/PageWithIntl";
import { Index } from "../../../pages/index";

// Mock the show service.
jest.mock("../../../app/service/showService");

// Mock the i18n local storage module.
jest.mock("../../../app/storage/i18n");

it("test page with i18n hoc component with the index page - snapshot test", () => {

    const IndexWithIntl = pageWithIntl(Index);

    const tree = renderer.create(
        <IndexWithIntl />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it("test page with i18n hoc component with the index page with context - snapshot test", () => {
    const context = {
        req: {
            headers: {
                cookie: "userLocale=fr;",
            },
        },
    };

    const IndexWithIntl = pageWithIntl(Index);
    IndexWithIntl.getInitialProps(context);

    const tree = renderer.create(
        <IndexWithIntl />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
