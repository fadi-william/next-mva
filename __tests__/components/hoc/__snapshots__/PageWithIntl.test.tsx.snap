// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test page with i18n hoc component with the index page - snapshot test 1`] = `
<div
  style={
    Object {
      "border": "1px solid #DDD",
      "margin": 20,
      "padding": 20,
    }
  }
>
  <div
    className="Header"
  >
    <nav>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <div
        className="lang-switcher-ph"
      >
        <select
          defaultValue="en"
          onChange={[Function]}
        >
          <option
            value="fr"
          >
            French
          </option>
          <option
            value="en"
          >
            English
          </option>
        </select>
      </div>
    </nav>
  </div>
  <div>
    <h1>
      Shows
    </h1>
    <div>
      Loading...
    </div>
  </div>
</div>
`;

exports[`test page with i18n hoc component with the index page with context - snapshot test 1`] = `
<div
  style={
    Object {
      "border": "1px solid #DDD",
      "margin": 20,
      "padding": 20,
    }
  }
>
  <div
    className="Header"
  >
    <nav>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <div
        className="lang-switcher-ph"
      >
        <select
          defaultValue="en"
          onChange={[Function]}
        >
          <option
            value="fr"
          >
            French
          </option>
          <option
            value="en"
          >
            English
          </option>
        </select>
      </div>
    </nav>
  </div>
  <div>
    <h1>
      Shows
    </h1>
    <div>
      Loading...
    </div>
  </div>
</div>
`;
